### PR TITLE
Updated retrieving MetaDataUrl from endpoint url

### PR DESCRIPTION
**Pull request by NicolajHansenPersonal**
Updated GetMetadataUrl to fetch the url from a simple request to the endpoint url.

Removed used of GetMetaDataUrl in SetCredentials because GetMetaDataUrl needs the credentials to function properly. Kept old way of find the metadataurl as a fallback

[AB#21786](https://dev.azure.com/dynamicwebsoftware/Dynamicweb/_workitems/edit/21786/)

### CODE REVIEW

**Comment by frederik5480** [comment]
I think it's odd the tabs is the length of 8 instead of 4, so when you format it, it makes it impossible to see the changes. If possible, please re-format it to the usual style of code, change your tab spacing to 4 if possible
image

**Review by MatthiasSort — APPROVED** [review]

**Comment by MatthiasSort** [comment]
@NicolajHansenPersonal remember to bump the version, else it will not be released to Nuget

**Comment by NicolajHansenPersonal** [comment]
> @NicolajHansenPersonal remember to bump the version, else it will not be released to Nuget

Just did :)

**Comment by NicolajHansenPersonal** [comment]
> I think it's odd the tabs is the length of 8 instead of 4, so when you format it, it makes it impossible to see the changes. If possible, please re-format it to the usual style of code, change your tab spacing to 4 if possible image

I don't know why it has marked the almost every line as something ive changed. But i can't seem to undo it :D

File: src/ODataProvider.cs [hunk 1]
```diff
@@ -129,6 +130,27 @@ public string DestinationEndpointId
     #endregion
 
     private string GetMetadataURL()
```
Comment (frederik5480) [inline src/ODataProvider.cs:132]: I think we can refactor this a bit, to something like

`    private string GetMetadataURL()
    {
        if (GetEndpointResponse(ODataSourceReader.GetEndpointUrlWithTop(_endpoint.Url), out string endpointResponse, out Exception exception) != HttpStatusCode.OK || exception is not null)
            return GetMetadataURLFallBack();

        using var responseJson = JsonDocument.Parse(endpointResponse);
        if (responseJson.RootElement.ValueKind != JsonValueKind.Object)
            return GetMetadataURLFallBack();

        return responseJson.RootElement.EnumerateObject().FirstOrDefault(obj => obj.Name.Equals("@odata.context", StringComparison.OrdinalIgnoreCase)).Value.GetString() ?? GetMetadataURLFallBack();
    }`

In general just avoiding the indentation, this also allows us to have more control of each fallback if we need to.

Comment (NicolajHansenPersonal) [reply src/ODataProvider.cs:132]: fixed

**Review by frederik5480 — APPROVED** [review]

**Review by MatthiasSort — APPROVED** [review]

**Pull request merged**